.github/workflows/ci.yml

```diff
@@ -91,6 +91,37 @@ jobs:
         with:
           name: Documentation
           path: _build/html
+  benchmarks:
+    defaults:
+      run:
+        shell: bash -el {0}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Cache conda
+        uses: actions/cache@v4
+        env:
+          # Increase this value to reset cache if ci/environment.yml has not changed
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key:
+            benchmarks-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/environment.yml') }}
+      - uses: conda-incubator/setup-miniconda@v3
+        with:
+          activate-environment: sparse-dev
+          allow-softlinks: true
+          environment-file: ci/environment.yml
+          python-version: '3.9'
+          miniforge-version: latest
+      - name: Install asv
+        run: |
+          pip install asv
+          asv machine --yes
+      - name: Run benchmarks
+        run: |
+          asv run
```

Comment:
```suggestion
          asv run --quick
```

Reply:
Done!

Reply:
@hameerabbasi Just to be sure - this option means that benchmarks in the PR aren't meaningful and they're only meant to make sure benchmarks can be run, right?

If someone actually wants to run benchmarks they need to do it locally.

Reply:
Correct, because the machine we'd get for benchmarks would be inconsistent anyway, so comparisons are useless here.

Even if it was consistent it's inside a VM on top of a system where other stuff is running, so there would be a lot of noise across CI runs.